### PR TITLE
CB-73: Generate API docs for Products

### PR DESCRIPTION
**Pull request by grabomarcin**
https://team-1626248039387.atlassian.net/browse/CB-73
I also made some bugfixes with other integration specs.

### CODE REVIEW

File: spec/integration/api/v1/products_spec.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  let(:product) { create(:product) }
+
+  path '/api/v1/products' do
+    get 'List Products' do
+      tags 'Products'
+      consumes 'application/json'
+      produces 'application/json'
+
+      response '200', 'returns list pf products' do
+        before { create_list(:product, 10) }
```
Comment (PatrykTyszka) [inline spec/integration/api/v1/products_spec.rb:12]: why not 100 :D .. :D 3 is enough

File: spec/integration/api/v1/products_spec.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      consumes 'application/json'
+      produces 'application/json'
+      parameter name: :product, in: :body, schema: {
+        type: :object,
+        properties: {
+          name: { type: :string }
+        }
+      }
+      response '200', 'creates and returns product' do
+        let(:product) { { product: { name: 'Water' } } }
```
Comment (PatrykTyszka) [inline spec/integration/api/v1/products_spec.rb:30]:
```suggestion
        let(:product) do
          { product: { name: 'Water' } }
        end
```

**Review by PatrykTyszka — APPROVED** [review]

**Review by Narzew — APPROVED** [review]

File: spec/integration/api/v1/products_spec.rb (hunk 1 above)
Comment (grabomarcin) [reply spec/integration/api/v1/products_spec.rb]: Fixed. :p

File: spec/integration/api/v1/products_spec.rb (hunk 2 above)
Comment (grabomarcin) [reply spec/integration/api/v1/products_spec.rb]: Fixed.

**Review by mirecki — APPROVED** [review]

**Pull request merged**